addons/web/tests/template_statusbar_time_testing.py

```diff
@@ -0,0 +1,170 @@
+            self.assertEqual(self.rec_2.json_statustime, track_json_statustime2)
+            self.assertEqual(self.rec_3.json_statustime, track_json_statustime3)
+
+            now.return_value += timedelta(minutes=10)
+            batch.stage_id = self.stage_1
+            _update_statustime_tracking(track_json_statustime1, [('-', self.stage_1), ('-', self.stage_2)])
+            _update_statustime_tracking(track_json_statustime2, [('-', self.stage_2)])
+            _update_statustime_tracking(track_json_statustime3, [('-', self.stage_2)])
+            self.assertFalse(self.rec_1.json_statustime)
+            self.assertFalse(self.rec_2.json_statustime)
+            self.assertFalse(self.rec_3.json_statustime)
+
+            now.return_value += timedelta(minutes=10)
+            batch.stage_id = self.stage_3
+            _update_statustime_tracking(track_json_statustime1, [('+', self.stage_1)])
+            _update_statustime_tracking(track_json_statustime2, [('+', self.stage_1)])
+            _update_statustime_tracking(track_json_statustime3, [('+', self.stage_1)])
+            self.assertEqual(self.rec_1.json_statustime, track_json_statustime1)
+            self.assertEqual(self.rec_2.json_statustime, track_json_statustime2)
+            self.assertEqual(self.rec_3.json_statustime, track_json_statustime3)
+
+            now.return_value += timedelta(minutes=10)
+            batch.stage_id = self.stage_2
+            self.assertEqual(self.rec_1.json_statustime, track_json_statustime1)
+            self.assertEqual(self.rec_2.json_statustime, track_json_statustime2)
+            self.assertEqual(self.rec_3.json_statustime, track_json_statustime3)
+
+            now.return_value += timedelta(minutes=10)
+            batch.stage_id = self.stage_4
+            _update_statustime_tracking(track_json_statustime1, [('+', self.stage_2)])
+            _update_statustime_tracking(track_json_statustime2, [('+', self.stage_2)])
+            _update_statustime_tracking(track_json_statustime3, [('+', self.stage_2)])
+            self.assertEqual(self.rec_1.json_statustime, track_json_statustime1)
+            self.assertEqual(self.rec_2.json_statustime, track_json_statustime2)
+            self.assertEqual(self.rec_3.json_statustime, track_json_statustime3)
+
+            now.return_value += timedelta(minutes=10)
+            batch.stage_id = self.stage_2
+            _update_statustime_tracking(track_json_statustime1, [('-', self.stage_2)])
+            _update_statustime_tracking(track_json_statustime2, [('-', self.stage_2)])
+            _update_statustime_tracking(track_json_statustime3, [('-', self.stage_2)])
+            self.assertEqual(self.rec_1.json_statustime, track_json_statustime1)
+            self.assertEqual(self.rec_2.json_statustime, track_json_statustime2)
+            self.assertEqual(self.rec_3.json_statustime, track_json_statustime3)
+
+    def template_test_queries_batch_json_statustime(self, number_of_queries):
```

Comment:
`number_of_queries` could be split into the different cases, so we understand what they mean when we are not on this page. So maybe a dictionary, or different named arguments.